src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -618,16 +608,15 @@ private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEta
                         continue;
                     }
 
-                    if (lastRaftIndex < maxEtag)
+                    if (maxEtag == -1 || lastRaftIndex < maxEtag)
                         maxEtag = lastRaftIndex.Value;
 
                     if (maxEtag == 0)
                         return CompareExchangeTombstonesCleanupState.NoMoreTombstones;
                 }
             }
 
-            if (state != null)
-            {
```

Comment:
This condition can be false only in tests.
I modified the tests to avoid that.
Also, I didn't revert the indentation so it will be easier to review the changes.
When approved, I can remove the indentation.